packages/create-webpack-app/src/utils/fileActions.ts

```diff
@@ -0,0 +1,221 @@
+  if (typeof content === "string") {
+    const identicalContent = await compareContentWithFile(renderedPath, content);
+    if (identicalContent) {
+      return { ...config, skip: () => true, operationType: "identical" };
+    }
+  }
+
+  const tempFilePath = path.join("/tmp", `temp_${path.basename(renderedPath)}`);
+  await fs.writeFile(tempFilePath, content || "");
+
+  try {
+    let userChoice: AddConfig | undefined;
+    while (!userChoice) {
+      const action = await expand({
+        message: `File conflict at ${path.basename(renderedPath)}?`,
+        choices: [
+          { key: "y", name: "overwrite", value: "overwrite" },
+          { key: "n", name: "do not overwrite", value: "skip" },
+          { key: "a", name: "overwrite this and all others", value: "overwrite_all" },
+          {
+            key: "d",
+            name: "Show the difference between the existing file and the new file",
+            value: "diff",
+          },
+          { key: "x", name: "abort", value: "abort" },
+        ],
+        expanded: false,
+      });
+
+      switch (action) {
+        case "overwrite":
+          userChoice = { ...config, operationType: "overwrite" };
+          break;
+        case "skip":
+          userChoice = { ...config, skip: () => true, operationType: "skip" };
+          break;
+        case "overwrite_all":
+          globalConfig.overwriteAll = true;
+          return { ...config, force: true, operationType: "overwrite" };
+        case "diff":
+          await getDiff(renderedPath, tempFilePath);
+          break;
+        case "abort":
+          logger.error("Aborting process...");
+          process.exit(1);
+      }
+    }
+    return userChoice;
+  } finally {
+    await fs.unlink(tempFilePath).catch(() => {
+      logger.warn(`Failed to delete temporary file: ${tempFilePath}`);
+    });
+  }
+}
+
+export default async function (plop: NodePlopAPI) {
+  plop.setPlopfilePath(path.resolve(__dirname, "../plopfile.js"));
+  plop.setDefaultInclude({ actions: true });
+
+  plop.setActionType("fileActions", async (answers, config) => {
+    const initialConfig = { ...config, data: answers, operationType: "create" } as AddConfig;
+    const finalConfig = await checkAndPrepareAction(initialConfig);
+
+    const renderedPath = finalConfig.path;
+    const operationType = finalConfig.operationType;
+    const templatePath = finalConfig.templateFile;
+
+    if (finalConfig.skip) {
+      return `${operationType}:${renderedPath}`;
+    }
+
+    const content = await renderTemplate(
+      finalConfig.template,
+      finalConfig.templateFile,
+      finalConfig.data,
+    );
+
+    if (!content) {
+      throw new Error("Template or templateFile is required");
+    }
+
+    await fs.mkdir(path.dirname(renderedPath), { recursive: true });
+
+    if (isBinaryFile(templatePath)) {
+      await fs.writeFile(renderedPath, content as Buffer);
+    } else {
+      await fs.writeFile(renderedPath, content as string, "utf8");
+    }
```

Comment:
You can just use `await fs.writeFile(renderedPath, content);`, node will do converting under the hood

Reply:
Earlier I was using that way, but then I was getting a problem where image wasnt loading and it turns out that, renderer was changing the content of image file and corrupting in the process. That's why I amhad to introduce this workaround

Reply:
I think you have  "utf8" before, just omit it when you write a file

Reply:
Yes, oh so if i dont specify it decides based on content?
Okay I will make this change,

Does readFile also work in similar ways?

Reply:
> Does readFile also work in similar ways?

If you have `"utf8"` in `readFile` you will get `string`, otherwise `Buffer`

Reply:
I tried doing it, the problem is, ejs.render first should be a string,

Reply:
Just convert it to string when you pass it to ejs (from `Buffer`), avoid extra check like `isBinaryFile`, it is wrong https://github.com/webpack/webpack-cli/pull/4249/files#diff-1554215a066d0b79435196cfc8e6329e9d77267b8e3780afa6176c35e4c39787R30 and

Reply:
Tried this but it is still causing the same corruption of .png file
```js
async function renderTemplate(
  template: string | undefined,
  templateFile: string | undefined,
  data: Answers | undefined
): Promise<Content | undefined> {
  if (template) {
    return ejs.render(template, data || {}, { async: true });
  }

  if (templateFile) {
    const fileContent: Content = await fs.readFile(templateFile);
    const templateContent = fileContent.toString("utf8");
    const renderedContent = await ejs.render(templateContent, data || {}, { async: true });
    return renderedContent;
  }

  return undefined;
}
```